src/renderer/task-runner.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  BisectRequest,
+  ElectronReleaseChannel,
+  RunResult,
+  RunnableVersion,
+  SetFiddleOptions,
+  SetupRequest,
+  TestRequest,
+} from '../interfaces';
+import { IpcEvents } from '../ipc-events';
+
+import { App } from './app';
+import { AppState } from './state';
+import { IpcRendererManager } from './ipc';
+import { Runner } from './runner';
+
+import { getVersionRange } from '../utils/get-version-range';
+import { normalizeVersion } from '../utils/normalize-version';
+
+export class TaskRunner {
+  private readonly autobisect: (v: RunnableVersion[]) => Promise<RunResult>;
+  private readonly done: (r: RunResult) => void;
+  private readonly hide: (channels: ElectronReleaseChannel[]) => Promise<void>;
+  private readonly log: (message: string) => void;
+  private readonly open: (o: SetFiddleOptions) => Promise<void>;
+  private readonly run: () => Promise<RunResult>;
+  private readonly setVersion: (ver: string) => Promise<void>;
+  private readonly show: (channels: ElectronReleaseChannel[]) => Promise<void>;
+
+  constructor(
+    app: App,
```

Comment:
`app` is also exposed via `window.ElectronFiddle.app`, which is usually used for one-off calls elsewhere in the code.

Reply:
e89536e82ff48932d1b142fa35b897735c7c7190

Reply:
...I spoke to soon. This `constructor()` function is called from App's `constructor()` function, meaning that the app instance isn't present in `window.ElectronFiddle.app`  yet. :roll_eyes:

Reply:
a607cac6a3079111430ef5412c260dc3f9ee0300 passes `App` in the constructor and then destructures that to get the state and runner objects. So the constructor is still relatively uncluttered but no longer relies on `window.ElectronFiddle.app`.